### PR TITLE
[MA-648] Add uninstall script for Module

### PR DESCRIPTION
**Pull request by mikeshutov-bolt**
# Description
Add uninstall script to clean up all Bolt created tables, columns and configurations when the Bolt plugin is uninstalled by a merchant.

Fixes: [MA-648](https://boltpay.atlassian.net/browse/MA-648)

#changelog [MA-648] Add uninstall script for Module

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Review by danac-gs — APPROVED** [review]

File: Setup/Uninstall.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    ){
+        $setup->startSetup();
+        // Remove Configurations 
+        $this->removeBoltConfig($setup);
+        
+        // Remove Bolt Tables
+        $tables = ['bolt_feature_switches',
+                   'bolt_external_customer_entity',
+                   'bolt_webhook_log',
+                   'bolt_customer_credit_cards'];
```
Comment (njfaries) [inline Setup/Uninstall.php:45]: If all the tables to remove have `bolt` in the name could we potentially get this list from the DB? Makes it so we don't have to update this if ever the DB changes or we add/remove tables.

File: Setup/Uninstall.php [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+     * @param SchemSetupInterface $setup
+     * 
+     * @return void
+     */
+    private function removeBoltColumns(SchemaSetupInterface $setup){
+        // Clean up quote table
+        $setup->getConnection()->dropColumn(
+            $setup->getTable('quote'),
+            'bolt_parent_quote_id'
+        );
```
Comment (njfaries) [inline Setup/Uninstall.php:93]: Same question as above, though I wonder if it's maybe more complex than it's worth here.

File: Setup/Uninstall.php [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+namespace Bolt\Boltpay\Setup;
+
+use Magento\Framework\Setup\ModuleContextInterface;
+use Magento\Framework\Setup\SchemaSetupInterface;
+use Magento\Framework\Setup\UninstallInterface;
+
+/**
+ * @codeCoverageIgnore
+ */
+class Uninstall implements UninstallInterface
```
Comment (njfaries) [inline Setup/Uninstall.php:27]: Is `UninstallInterface` a magento thing?

Comment (mikeshutov-bolt) [reply Setup/Uninstall.php:27]: Yea thats why we:
`use Magento\Framework\Setup\UninstallInterface;`

File: Setup/Uninstall.php (hunk 2 above)
Comment (mikeshutov-bolt) [reply Setup/Uninstall.php:93]: I think this one might be more effort than it's worth at the moment but the table one makes sense and is more likely to grow as we add features into the plugin so I will take a look at that one.

**Review by njfaries — APPROVED** [review]

**Pull request merged**